packages/devtools/lib/src/timeline/timeline.dart

```diff
@@ -236,6 +258,63 @@ class TimelineScreen extends Screen {
     _updateListeningState();
   }
 
+  void loadFromImport(Map<String, dynamic> json) {
+    pauseButton.attribute('hidden', true);
+    resumeButton.attribute('hidden', true);
+    clearButton.attribute('hidden', true);
+    exportButton.attribute('hidden', true);
+    exitImportModeButton.attribute('hidden', false);
+
+    final List<dynamic> events = json['traceEvents'] ?? [];
+    final traceEvents = events
+        .cast<Map<String, dynamic>>()
+        .map((trace) => TraceEvent(trace))
+        .toList();
+    final Map<String, dynamic> cpuProfile = json['cpuProfile'] ?? {};
+
+    if (traceEvents.isEmpty && cpuProfile.isEmpty) {
+      framework.toast('Imported file does not contain timeline data.');
+      _exitImportMode();
+      return;
+    }
+
+    _clearTimeline();
+
+    if (traceEvents.isNotEmpty) {
+      timelineController.loadTimelineFromImport(traceEvents, cpuProfile);
+    } else {
+      framesBarChart.attribute('hidden', true);
+    }
+
+    if (cpuProfile.isNotEmpty) {
+      // TODO(kenzie): load CPU profile.
+    } else {
+      _destroySplitter();
+    }
+  }
+
+  void _exitImportMode() {
+    pauseButton.attribute('hidden', false);
```

Comment:
lets add a top level field to the framework indicating we are in _offlineMode and use it here as well as other places in the inspector in the future.
All this logic could then be keyed on updating the view to reflect that the state is _offlineMode.